### ISSUE TITLE
Types of property 'cronTime' are incompatible.

### ISSUE THREAD

**Issue by andrey-o-first-am** [nestjs/schedule#1640]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

   "nest": "10.3.2"
   "@nestjs/schedule": "^4.0.1"
   "@types/cron": "^2.4.0"
   "cron": "^3.1.7",

```
         const cronJob = new CronJob(
          '* * * * *',
          () => {
            this.logger.log(`${eventName} cron executed`);
          },
          null,
          false,
          'America/Los_Angeles',
        );
        
        this.schedulerRegistry.addCronJob(eventName, cronJob); // <- ts error here
        cronJob.start();
        
```

Error:
```
Argument of type 'import("/Users/aorlov/Projects/properti-api/node_modules/cron/dist/job").CronJob<null, null>' is not assignable to parameter of type 'import("/Users/aorlov/Projects/properti-api/node_modules/@nestjs/schedule/node_modules/cron/dist/job").CronJob<null, null>'.
  Types of property 'cronTime' are incompatible.
    Type 'import("/Users/aorlov/Projects/properti-api/node_modules/cron/dist/time").CronTime' is not assignable to type 'import("/Users/aorlov/Projects/properti-api/node_modules/@nestjs/schedule/node_modules/cron/dist/time").CronTime'.
      Types have separate declarations of a private property 'second'.ts(2345)
      
```

### Minimum reproduction code

https:github.com

### Steps to reproduce

```       
 const cronJob = new CronJob(
          '* * * * *',
          () => {
            this.logger.log(`${eventName} cron executed`);
          },
          null,
          false,
          'America/Los_Angeles',
        );
        
        this.schedulerRegistry.addCronJob(eventName, cronJob);
        cronJob.start();
      
 ```

### Expected behavior

The cronJob singleton should be able to be accepted by the addCronJob method as a parameter without any typescript errors. 

### Package version

4.0.1

### NestJS version

10.3.2

### cron version
3.1.3 

### Node.js version

v21.2.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

cron": "^3.1.7

**Comment by llllvvuu** [comment]
This will be fixed by #1633

NestJS practice of using exact dependencies instead of `^` makes it very susceptible to importing two different versions of a package.

**Comment by kamilmysliwiec** [comment]
https://github.com/nestjs/schedule/pull/1633
